examples/magento-graphcms/pages/sitemap/content.xml.tsx

```diff
@@ -0,0 +1,72 @@
+import { getHygraphStaticPaths } from '@graphcommerce/graphcms-ui'
+import { getCategoryStaticPaths } from '@graphcommerce/magento-category/queries/getCategoryStaticPaths'
+import { canonicalize } from '@graphcommerce/next-ui'
+import { GetServerSideProps } from 'next'
+import { ISitemapField, getServerSideSitemapLegacy } from 'next-sitemap'
+import { graphqlSsrClient } from '../../lib/graphql/graphqlSsrClient'
+import { isMatch } from '../../lib/sitemap/matcher'
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const { locale, res, defaultLocale } = context
+
+  if (!locale) throw Error('Locale not found')
+
+  const additionalPaths = ['']
+  const excludes = [
+    '*page/home',
+    '*/p/*',
+    '*/product/global',
+    '*/product*',
+    '*/account*',
+    '*/wishlist*',
+    '*/cart*',
+    '*/modal*',
+    '*/checkout*',
+    '*/test/*',
+    '/test/*',
+    '*/404',
+    '*/no-route',
+    '*/home',
+    '*/switch-stores',
+    '*/search',
+    '*/account',
+    '*/wishlist',
+    '*/cart',
+    '*/checkout',
+  ]
+
+  // Filter out category pages as they are already added in sitemap/categories.xml
+  excludes.push(
+    ...(await getCategoryStaticPaths(graphqlSsrClient(locale), locale, { limit: false })).map(
+      (path) => `/${typeof path === 'string' ? path : path.params.url.join('/')}`,
+    ),
+  )
```

Comment:
As discussed earlier, might it be possible to get rid of these excludes, this will make it much slower.

Reply:
This query takes around half a second for a fairly large category tree (uncached - cached is obviously a lot faster).
Seeing as this page is intended for bots only and [bots don't seem to care](https://www.lumar.io/blog/best-practice/google-times-out-after-two-minutes-when-crawling-sitemaps/) about a few seconds more or less, I'd suggest keeping this in to prevent duplicate URLs.